lib/transmuxer.js

```diff
@@ -993,11 +1036,61 @@ VideoSegmentStream = function(track) {
     boxes.set(moof);
     boxes.set(mdat, moof.byteLength);
 
-    this.trigger('data', boxes);
+    clearDTSInfo(track);
+    this.trigger('data', {track: track, boxes: boxes});
   };
 };
 VideoSegmentStream.prototype = new muxjs.Stream();
 
+collectDTSInfo = function (track, data) {
+  if (track.minDTS === undefined) {
+    track.minDTS = data.dts;
+  } else {
+    track.minDTS = Math.min(track.minDTS, data.dts);
+  }
+
+  if (track.maxDTS === undefined) {
+    track.maxDTS = data.dts;
+  } else {
+    if (data.dts - track.maxDTS > 100) {
+      track.deltaDTS = data.dts - track.maxDTS;
+    }
+    track.maxDTS = Math.max(track.maxDTS, data.dts);
+  }
+};
+
+clearDTSInfo = function (track) {
+  delete track.minDTS;
+  delete track.maxDTS;
+  delete track.deltaDTS;
+};
+
+calculateTrackBMDT = function (track) {
+  var scale;
+  var nextExpectedBMDT = track.nextExpectedBMDT;
```

Comment:
One `var` per line unless you want to put together a patch to update the style across the whole project.